app/actions/websocket/teams.ts

```diff
@@ -49,3 +52,90 @@ export async function handleLeaveTeamEvent(serverUrl: string, msg: any) {
         }
     }
 }
+
+export async function handleUpdateTeamEvent(serverUrl: string, msg: WebSocketMessage) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    const teamFromData = JSON.parse(msg.data.team);
+    const team: Team = {
+        ...teamFromData,
+    };
+
+    database.operator.handleTeam({
+        teams: [team],
+        prepareRecordsOnly: false,
+    });
+}
+
+export async function handleTeamAddedEvent(serverUrl: string, msg: WebSocketMessage) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    const user = await queryCurrentUser(database.database);
+    if (!user) {
+        return;
+    }
+
+    const [team, memberships] = await Promise.all([
+        fetchMyTeam(serverUrl, msg.data.team_id),
+    ]);
+
+    console.log('team', team);
+    console.log('memberships', memberships);
+
+    // const [team, member, teamUnreads] = await Promise.all([
+    //     Client4.getTeam(msg.data.team_id),
+    //     Client4.getTeamMember(teamId, userId),
+    //     Client4.getMyTeamUnreads(),
+    // ]);
+
+    // const actions = [];
+    // if (team) {
+    //     actions.push({
+    //         type: TeamTypes.RECEIVED_TEAM,
+    //         data: team,
+    //     });
+    //
+    //     if (member) {
+    //         actions.push({
+    //             type: TeamTypes.RECEIVED_MY_TEAM_MEMBER,
+    //             data: member,
+    //         });
+    //
+    //         if (member.roles) {
+    //             const rolesToLoad = new Set<string>();
+    //             for (const role of member.roles.split(' ')) {
+    //                 rolesToLoad.add(role);
+    //             }
+    //
+    //             if (rolesToLoad.size > 0) {
+    //                 const roles = await Client4.getRolesByNames(Array.from(rolesToLoad));
+    //                 if (roles.length) {
+    //                     actions.push({
+    //                         type: RoleTypes.RECEIVED_ROLES,
+    //                         data: roles,
+    //                     });
+    //                 }
+    //             }
+    //         }
+    //     }
+    //
+    //     if (teamUnreads) {
+    //         actions.push({
+    //             type: TeamTypes.RECEIVED_MY_TEAM_UNREADS,
+    //             data: teamUnreads,
+    //         });
+    //     }
+    // }
+    //
+    // if (actions.length) {
+    //     dispatch(batchActions(actions, 'BATCH_WS_TEAM_ADDED'));
+    // }
+    //
+    // return {data: true};
```

Comment:
??

Reply:
@enahum this commented section is what I will be handling in a subsequent commit.  This is the V1 code that I'll be converting. From my current understanding, each `action` will need to be converted to a WatermelonDB operation.  From my understanding some of these handle functions are available and I've added them as comments above.